lib/src/upZenohClient.cpp

```diff
@@ -28,9 +28,31 @@
 using namespace uprotocol::v1;
 using namespace uprotocol::client;
 
-std::shared_ptr<UpZenohClient> UpZenohClient::instance(void) noexcept {
-    static std::weak_ptr<UpZenohClient> w_handle;
+static std::weak_ptr<UpZenohClient> w_handle;
 
+std::shared_ptr<UpZenohClient> UpZenohClient::instance() noexcept {
+
+    if (auto handle = w_handle.lock()) {
+        return handle;
+    } else {
+        static std::mutex construction_mtx;
+        std::lock_guard lock(construction_mtx);
+
+        if (handle = w_handle.lock()) {
+            return handle;
+        }
+
+        handle = std::make_shared<UpZenohClient>(ConstructToken());
+        if (handle->rpcSuccess_.code() == UCode::OK && handle->uSuccess_.code() == UCode::OK) {
+            w_handle = handle;
+            return handle;
+        } else {
+            return nullptr;
+        }
+    }
+}
+
+std::shared_ptr<UpZenohClient> UpZenohClient::instance(const uprotocol::v1::UAuthority& src_authority) noexcept {
```

Comment:
Instead of copy-pasting code, could this be streamlined with std::optional?

```suggestion
std::shared_ptr<UpZenohClient> UpZenohClient::instance() noexcept {
    return instance(std::nullopt_t);
}

std::shared_ptr<UpZenohClient> UpZenohClient::instance(std::optional<const uprotocol::v1::UAuthority&> src_authority) noexcept {
    static std::weak_ptr<UpZenohClient> w_handle;
```